src/OrchardCore.Modules/OrchardCore.Security/Services/ContentSecurityPolicyHeaderPolicyProvider.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace OrchardCore.Security.Services
+{
+    public class ContentSecurityPolicyHeaderPolicyProvider : HeaderPolicyProvider
+    {
+        public override void ApplyPolicy(HttpContext httpContext)
+        {
+            if (Options.ContentSecurityPolicy.Length > 0)
+            {
+                httpContext.Response.Headers[SecurityHeaderNames.ContentSecurityPolicy] = String.Join(SecurityHeaderDefaults.PoliciesSeparator, Options.ContentSecurityPolicy);
```

Comment:
I think we should do some pre-processing to not have to do a `String.Join()` on maybe many strings, this on each request when the middleware call the `ApplyPolicy()` of providers

Reply:
Removing the 'None` policies should fix this issue

Reply:
Hmm, still worth to do some pre-processing, e.g. could be done by the provider ctors, so that when building a given provider you could pass the options to its ctor that will do e.g. the join.

Need to leave, will see tomorrow ;)

Reply:
Please check the latest changes, I remove them in the driver, so no preprocessing are required here